### PR TITLE
🪟 fix+feat: General UI Enhancements

### PR DESCRIPTION
**Pull request by techwithanirudh**
## Summary

A few UI changes to match closer to OpenAI's design and a few general UI Updates.

Before:
image

After:
image

Before:
image

After:
image

Before:
image

After:
image

Before:
image

After:
image

Before:
image

After:
image

Before:
image

After:
image

Before:
image

After:
image

Before:
image

After:
image

### General Fixes:
fix: Change validation errors texts from black/white to red.
fix: Handle an edge case where focus is lost while a SocialButton is held, but not clicked
fix: Add focus border for ChatInput on light mode
feat: Add disabled property for NavLink
feat: Move export conversation top right, like ChatGPT's share button.
image
feat: Match Login/Register/ResetPassword Form input styles to OpenAI Input styles
fix: Optimize settings for mobile, and add a fixed width for the modal
fix: General colors fixes for dark mode
fix: Hover states for Prev/Next buttons

## Change Type

- [x] Bug fix (non-breaking change which fixes an issue)

## Testing

N/A

### **Test Configuration**:

N/A

## Checklist

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented in any complex areas of my code
- [x] My changes do not introduce new warnings
- [x] Local unit tests pass with my changes

### CODE REVIEW

**Comment by danny-avila** [comment]
There's 2 minor merge conflicts, I will see if I can resolve them later unless you beat me to it.

**Comment by techwithanirudh** [comment]
@danny-avila I resolved the merge conflicts 😉

**Comment by danny-avila** [comment]
A lot of changes here that could've been individual PRs. Please reduce the scope of your PRs next time.

File: client/src/components/Chat/Header.tsx [hunk 1]
```diff
@@ -17,17 +26,67 @@ export default function Header() {
...
+
   return (
     <div className="sticky top-0 z-10 flex h-14 w-full items-center justify-between bg-white p-2 font-semibold dark:bg-gray-800 dark:text-white">
-      <div className="hide-scrollbar flex items-center gap-2 overflow-x-auto">
-        {!navVisible && <HeaderNewChat />}
-        {interfaceConfig.endpointsMenu && <EndpointsMenu />}
-        {modelSpecs?.length > 0 && <ModelSpecsMenu modelSpecs={modelSpecs} />}
-        {<HeaderOptions interfaceConfig={interfaceConfig} />}
-        {interfaceConfig.presets && <PresetsMenu />}
+      <div className="hide-scrollbar flex w-full items-center justify-between gap-2 overflow-x-auto">
```
Comment (danny-avila) [inline client/src/components/Chat/Header.tsx:23]: Refactor the export functions into its own component, please.

**Review by danny-avila — CHANGES_REQUESTED** [review]

File: client/src/components/Chat/Header.tsx (hunk 1 above)
Comment (danny-avila) [reply client/src/components/Chat/Header.tsx:23]: this change should only add 2 lines to `client/src/components/Chat/Header.tsx`

**Review by danny-avila — APPROVED** [review]

**Pull request merged**